lib/generator.js

```diff
@@ -269,25 +279,33 @@ const generateMd = (inventory, options = {}) => {
       buf.push(headerify(`Interface: ${name}\n`, options.interfaceLevel));
     }
     for (const [method, signature] of methods) {
-      const signatures = [];
+      const defaultArgs = signature.parameters
+        .filter(p => p.offset === 2)
+        .map(p => p.name);
+      let signatures = [defaultArgs];
       if (signature.argsSignature) {
-        signature.argsSignature.forEach(s =>
-          signatures.push(s.replace(/\s/g, '').split(','))
-        );
-      } else if (signature.parameters.length) {
-        const defaultArgs = signature.parameters
-          .filter(p => p.offset === 2)
-          .map(p => p.name);
-        signatures.push(defaultArgs);
-      } else {
+        signatures = [];
+        let compliant = false;
+        signature.argsSignature.forEach(s => {
+          const argsSignature = s.replace(/\s/g, '').split(',');
+          signatures.push(argsSignature);
+          if (checkArgsSignature(argsSignature, defaultArgs)) compliant = true;
+        });
+        if (!compliant) {
+          console.warn(
+            `\x1b[33mWarning: No signatures for function '${method}' ` +
```

Comment:
I don't think that using ANSI escape codes will work in every Unix terminal or on Windows systems, I think you should check if the terminal supports them and use them only in that case.

Reply:
IMO, it will be easier to not support colours at all for now. And if we were to support them in the future that's an issue for another PR.